src/mlpack/methods/decision_tree/gini_gain.hpp

```diff
@@ -156,6 +163,7 @@ class GiniGain
     {
       // SIMD loop: add counts for four elements simultaneously (if the compiler
       // manages to vectorize the loop).
+      // #pragma omp parallel for reduction (+:counts)
```

Comment:
I think we can modify reduction definition to work on arma data-types. If you face any issue, I would love to help with same.

Reply:
Hello, I would love to work on the same. Once I am done working on this issue, I'll start thinking how do i modify reduction. This would surely help in Profiling for Parallelization project for GSoC 2020.

Reply:
Great.

Reply:
Hey Param, Could you take  a look at these links, https://software.intel.com/en-us/articles/openmp-40-new-features-supported-in-intel-compiler-160 and https://gist.github.com/eruffaldi/7180bdec4c8c9a11f019dd0ba9a2d68c . Since there are some implementations to support reduction for vectors / arrays, maybe we could extend them to armadillo. Not sure though. Let me know what you think. Thanks